### PR TITLE
Refactor modules to stick to the specification

### PR DESCRIPTION
**Pull request by hansl**
Notably async module and ledger transactions/commands. This has been tested
with a special PR in framework that will be updated soon in its own PR.

### CODE REVIEW

File: src/many/Cargo.toml [hunk 1]
```diff
@@ -15,7 +15,7 @@ base32 = "0.4.0"
 base64 = "0.13.0"
 cbor-diag = "0.1.8"
 crc-any = "2.4.0"
-cryptoki = "0.3"
+cryptoki = { version = "0.3", features = ["generate-bindings"] }
```
Comment (fmorency) [inline src/many/Cargo.toml:18]: Reason for enabling this?

File: src/many/src/server/module/_5_kvstore_commands.rs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+use crate::{Identity, ManyError};
```
Comment (fmorency) [inline src/many/src/server/module/_5_kvstore_commands.rs:1]: `kvstore_commands` is `7` in `main`

File: src/many/src/server/module/_5_kvstore_commands.rs [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+use many_macros::many_module;
+
+mod put;
+pub use put::*;
+
+#[many_module(name = KvStoreCommandsModule, id = 5, namespace = kvstore, many_crate = crate)]
+pub trait KvStoreCommandsModuleBackend: Send {
+    fn put(&mut self, sender: &Identity, args: PutArgs) -> Result<PutReturns, ManyError>;
```
Comment (fmorency) [inline src/many/src/server/module/_5_kvstore_commands.rs:9]: `delete` is missing

File: src/many/src/server/module/_8_async.rs [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+}
+
+#[derive(Debug, Clone, Encode, Decode)]
+#[cbor(map)]
+pub struct StatusArgs {
+    #[n(0)]
+    pub token: AsyncToken,
+}
+
+pub enum RequestStatus {
```
Comment (fmorency) [inline src/many/src/server/module/_8_async.rs:111]: Where is this used? I don't see this in the spec.

File: src/many/src/server/module/_8_async.rs [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+    Queued,
+    Processing,
+    Done { response: CoseSign1 },
+    Expired,
+}
+
+impl Encode for StatusReturn {
+    fn encode<W: Write>(&self, e: &mut Encoder<W>) -> Result<(), Error<W::Error>> {
+        match self {
+            StatusReturn::Unknown => e.map(1)?.u8(0)?.u8(0)?,
```
Comment (fmorency) [inline src/many/src/server/module/_8_async.rs:129]: Use an `enum` value instead of hardcoding the return value ID?

**Review by fmorency — CHANGES_REQUESTED** [review]
LGTM, except the small changes and questions.

File: src/many/Cargo.toml (hunk 1 above)
Comment (hansl) [reply src/many/Cargo.toml:18]: Doesn't build on ARM CPUs. Level2 archs and stuff.

Comment (hansl) [reply src/many/Cargo.toml:18]: I _could_ special case it for arm archs if you want. Does that cost you a lot of compile time?

File: src/many/src/server/module.rs [hunk 6]
```diff
@@ -5,24 +5,27 @@ use async_trait::async_trait;
...
-    _1_blockchain = blockchain;
-    _2_ledger = ledger;
-    _3_kvstore = kvstore;
-    _4_ledger_transactions = ledger_transactions;
-    _1000_abci_backend = abci_backend;
-    _1001_abci_frontend = abci_frontend;
+    base: _0_base;
+    blockchain: _1_blockchain;
+    ledger: _2_ledger + _4_ledger_transactions + _6_ledger_commands;
+    kvstore: _3_kvstore + _5_kvstore_commands;
```
[the remark this answers is not in the document]
Comment (hansl) [reply src/many/src/server/module.rs:25]: You're right. Done.

File: src/many/src/server/module/_8_async.rs (hunk 4 above)
Comment (hansl) [reply src/many/src/server/module/_8_async.rs:111]: It's here: https://github.com/many-protocol/specification/blob/main/attributes/network/cddl/8_async.cddl#L12

Comment (hansl) [reply src/many/src/server/module/_8_async.rs:111]: Removed.

File: src/many/src/server/module/_8_async.rs (hunk 5 above)
Comment (hansl) [reply src/many/src/server/module/_8_async.rs:129]: Can't use enum values with variants. But I isolated getting the variant id in a separate private function.

Comment (fmorency) [reply src/many/src/server/module/_8_async.rs:129]: Would it make sense to use something like https://crates.io/crates/num_enum? Maybe it's overkill, idk.

**Comment by hansl** [comment]
@fmorency sorry I lost myself in some monads regarding decoding StatusReturn. PTAL now.

**Review by fmorency — APPROVED** [review]

**Pull request merged**